### DISCUSSION TITLE
New strategy for information density

### DISCUSSION THREAD

**Discussion by Alkarex** [FreshRSS/FreshRSS#5900] [Strategy]
> I'd like to make it the default

_Originally posted by @Frenzie in https://github.com/FreshRSS/FreshRSS/issues/5899#issuecomment-1823134419_

@Frenzie @math-GH I have been increasingly thinking that we should change the views we have by:
1. Merge our *normal* and *reader* views
2. Instead, make a new selector (or two - for the vertical and horizontal preferences) to tune the information density:
    * a. when the information density increases horizontally, at some point use the same *aside feeds* logic than what we have for the mobile view (i.e. collapsed)
        * https://github.com/FreshRSS/FreshRSS/issues/6119
    * b. the reader view is replaced by the behaviour of the lowest vertical density (articles fully opened by default)
    * d. some of the steps for the vertical density would include allowing two or more lines for the titles, and more or less text preview
        * https://github.com/FreshRSS/FreshRSS/issues/6121
    * e. some of the steps of the horizontal density could control displaying thumbnails or not, just like some of the other columns (in addition to the above-mentioned *aside feeds*)
3. remove the options for choosing to have the articles opened or closed by default (contributes to simplifying the options), as this becomes redundant with the vertical density selector
4. Keep the preference for the selector level locally on the device, so it is possible to have two different settings for the same account used e.g. on mobile phone, tablet, computer. One setting could be *Auto* with a heuristic based on window size.
    * https://github.com/FreshRSS/FreshRSS/issues/5899)

An alternative to controlling the horizontal and vertical densities independently would be to offer a few predefined combinations.

In terms of implementation, I do not expect many changes, as it is mostly reworking slightly (and cleaning) what we already have.

Thoughts?

**Comment by Frenzie** [comment]
You could solve the title problem while reworking everything, but to my mind this is almost completely unrelated. :-)

> Merge our normal and reader views

I always thought it made sense to have them as separate views to be able to quickly switch between them. That is to say, in your concept I think you'd want to be able to define presets (and maybe assign them to particular feeds or categories?).

Even if the preset is just for "save how I have things set up right now so I can experiment more freely".

> remove the options for choosing to have the articles opened or closed by default (contributes to simplifying the options), as this becomes redundant with the vertical density selector

I imagine it can still be useful to collapse something even with expanded by default, but I don't know if people who use expanded by default do that in practice.

**Reply by Alkarex** [reply]
> I imagine it can still be useful to collapse something even with expanded by default

Completely agree: we should keep the manual ability to open/close articles. I was referring to the settings / options in FreshRSS configuration.

**Comment by math-GH** [comment]
> Merge our normal and reader views
I would suggest to make a survey, similar to #4295
I do not have any overview how often the global and reader view is used. Let's ask the community

**Reply by Alkarex** [reply]
The merge would be technical. At the moment, there is a lot of duplicated code. The actual behaviour of the normal and reader views would still be achievable with the new approach. So I am not sure what the survey could tell.
Edit: for the record, I personally use both, probably around 85% normal view, 15% reader view, 0% global view.

**Comment by math-GH** [comment]
I would like to develop a "Zen" mode, that hides a lot of button elements so the reader can focus on the articles.
I will discuss the idea later after finishing https://github.com/FreshRSS/FreshRSS/pull/6047
